components/Input.tsx

```diff
@@ -1,41 +1,126 @@
 import Image from "next/image";
 import { useState } from "react";
+import {
+  FieldErrors,
+  FieldValues,
+  UseFormGetValues,
+  UseFormRegister,
+  UseFormSetError,
+} from "react-hook-form";
+import { postEmailIsValid } from "@/lib/api";
 import eyeOff from "@/public/eye-off.png";
 import eyeOn from "@/public/eye-on.png";
+import { InputType } from "@/types/commonTypes";
 import styles from "./Input.module.css";
 
-function Input() {
-  const [isError, setIsError] = useState(false);
-  const [isEyeOn, setIsEyeOn] = useState(false);
+interface Props {
+  placeholder: string;
+  type: InputType;
+  register: UseFormRegister<FieldValues>;
+  errors: FieldErrors<FieldValues>;
+  setError: UseFormSetError<FieldValues>;
+  getValues: UseFormGetValues<FieldValues>;
+}
+
+const options = {
+  signInEmail: {
+    required: "이메일을 입력해주세요",
+  },
+  signInPassword: {
+    required: "비밀번호를 입력해주세요",
+  },
+  signUpEmail: {
+    required: "이메일을 입력해주세요",
+    validate: {
+      isValidEmail: async (email: FieldValues) => {
+        try {
+          const response = await postEmailIsValid({ email });
+          return response && true;
```

Comment:
이 부분만 본다면, 이렇게 작성할 수 있습니다.
```suggestion
          return !!response;
```